### PR TITLE
Jake Heft

### PR DESCRIPTION
**Pull request by jakeheft**
This PR will:
- Add in all classes and test classes
- Pass all tests, with the exception of museum lottery test

### CODE REVIEW

File: lib/museum.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    interested_patrons = {}
+    @exhibits.each do |exhibit|
+      @patrons.each do |patron|
+        if interested_patrons[exhibit] && patron.interests.include?(exhibit.name)
+          interested_patrons[exhibit] << patron
+        elsif patron.interests.include?(exhibit.name)
+          interested_patrons[exhibit] = [patron]
+        elsif !interested_patrons[exhibit] && !patron.interests.include?(exhibit.name)
+          interested_patrons[exhibit] = []
+        end
```
Comment (megstang) [inline lib/museum.rb:34]: oh nice! This would have solved that issue I was having in the review today. Good call on adding this in to catch this edge case.

File: test/museum_test.rb [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    @dmns.add_exhibit(@imax)
+    patron_1.add_interest("Gems and Minerals")
+    patron_1.add_interest("Dead Sea Scrolls")
+    patron_2.add_interest("Dead Sea Scrolls")
+    patron_3.add_interest("Dead Sea Scrolls")
+    @dmns.admit(patron_1)
+    @dmns.admit(patron_2)
+    @dmns.admit(patron_3)
+
+    @dmns.stubs(:draw_lottery_winner).returns(patron_1.name)
```
Comment (megstang) [inline test/museum_test.rb:93]: 👍

File: test/museum_test.rb [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+    assert_equal [patron_1, patron_3], @dmns.ticket_lottery_contestants(@dead_sea_scrolls)
+    assert_equal "Bob", @dmns.draw_lottery_winner(@dead_sea_scrolls)
+    assert_nil @dmns.draw_lottery_winner(@gems_and_minerals)
```
Comment (megstang) [inline test/museum_test.rb:97]: This may have a hard time passing later since you've already stubbed out the draw_lottery_winner method earlier. You may consider moving this assertion before you stub out that method.

**Review by megstang — COMMENTED** [review]
@jakeheft I know you weren't able to get through all of I3, but I think the code you have so far looks really good. Your testing for the last portion of I3 looks good other than where you placed your stub. Take a look at that-- it may cause some issues since you stubbed the method before both of your assertions for that final method.
